dagshub/data_engine/model/datasource.py

```diff
@@ -343,6 +345,31 @@ def send_datapoints_to_annotation(self, datapoints: Union[List[Datapoint], List[
             webbrowser.open_new_tab(link)
         return link
 
+    def wait_until_ready(self, max_wait_time=None):
+        """
+       Blocks until the datasource preprocessing is complete
+
+       Args:
+           max_wait_time (int): Maximum time to wait in seconds
+       """
```

Comment:
I'm not sure if having this thing indefinite by default is normal. I think we should set it to something substantial enough that a reasonable sized repo could be scanned.
10 minutes or so maybe

If the processing gets stuck at any point you don't want the users to just be stuck forever

Reply:
Ask Simon what he thinks a reasonable default timeout would be